### ISSUE TITLE
Prop has error type in template when it is a generic type which extends other type.

### ISSUE THREAD

**Issue by FlareZh** [vuejs/language-tools#3654]
This is the error:
![image](https://github.com/vuejs/language-tools/assets/22744623/35467cac-04b5-4a5b-9d12-176c0def3248)

Reproducible:
```vue
<script setup lang="ts" generic="T extends number">
const props = withDefaults(defineProps<{
	value?: T;
}>(), {
	value: 1 as any,
  // value: () => 1 as T
});
props.value.toFixed(2)
</script>

<template>
  {{ value.toFixed(2) }}
  {{ props.value.toFixed(2) }}
  {{ $props.value.toFixed(2) }}
</template>
```
[link](https://play.vuejs.org/#__SSR__eNqNU8uO2zAM/BVC6CEBEgvt9uQm6faxBdpDu2hz1MVrM462tiRIdJIi8L+Xlt3EzT6wvpFDijND+ig+OJfsGhSpWITca0cQkBoHVWbKpRIUlIASDXqdc7gGPBCaIoBp6jv0SqyUya0JBM5bF2AJe03bz7jJmorCpMCNNnjbQYujMop2WdXg+xTW75RpV5PpDM7pFF5DFiAzf2bKAEgJQ3oyheWqB9fcNuXeOC2JeEL2iz5gMXkzVWYhexVMiwPC2lUZIUcAx2P/3qge2nZAHn/vjL96pmAhT3PETOjaWU/zOnPJfbCGjWWFAGoA2M+00wz8KcHOd7ESWyIXUinzwnBbgZXe+cQgSeNqec1l0jeGdI3zwtbXV8lV8lYWOtA4nWCo53fe7gN6fkSJaOMwRnJyh37ueXvou829bOxF23j0BfRgfDed7WnZFAp8JBtdXliS29rpCv0PR5qP6D9rsqqy+28xR77Bk5Z8i/nvR/L34dBruvUYmY30U+ZLpB6++fWdb3gE1rZoqmENT4A/Mdiq6Tj2ZR8bUzDtUV1k+zVuWJtyHW663yT8E9URjW7E+riPT89IP9Nlt08utn8B0Ww+ZQ==)

**Comment by johnsoncodehk** [comment]
Fixed by #3636, you can use `props.value?.toFixed(2)` after v1.8.17.

**Comment by FlareZh** [comment]
> Fixed by #3636, you can use `props.value?.toFixed(2)` after v1.8.17.

Is there any plan to fix it when using `value.toFixed(2)`?

**Comment by FlareZh** [comment]
> This is a type issue with `defineComponent()`, please report it to the core repo.

Ok, Thank you!
